src/main/java/com/microsoft/sqlserver/jdbc/SharedTimer.java

```diff
@@ -56,16 +59,17 @@ static synchronized boolean isRunning() {
      *
      * If the reference count reaches zero then the underlying executor will be shutdown so that its thread stops.
      */
-    public synchronized void removeRef() {
-        if (refCount <= 0) {
-            throw new IllegalStateException("removeRef() called more than actual references");
-        }
-        refCount -= 1;
-        if (refCount == 0) {
-            // Removed last reference so perform cleanup
-            executor.shutdownNow();
-            executor = null;
-            instance = null;
+    public void removeRef() {
```

Comment:
`removeRef()` doesn't have to be `public`.

Reply:
The class 'SharedTimer' is a package level class, these methods are not accessible to public eitherway. I didn't modify any implementation so this stayed.